modules/gapi/src/backends/ie/giebackend.cpp

```diff
@@ -563,29 +726,28 @@ struct Infer: public cv::detail::KernelTag {
         return result;
     }
 
-    static void run(IECompiled &iec, const IEUnit &uu, IECallContext &ctx) {
+    static void run(IECompiled                            &iec,
+                    const IEUnit                          &uu,
+                    std::shared_ptr<IECallContext>        &&ctx,
+                    cv::gimpl::GIslandExecutable::IOutput &out,
+                    cv::gimpl::ie::SyncPrim               &sync) {
         // non-generic version for now:
         // - assumes all inputs/outputs are always Mats
         Views views;
         for (auto i : ade::util::iota(uu.params.num_in)) {
             // TODO: Ideally we shouldn't do SetBlob() but GetBlob() instead,
             // and redirect our data producers to this memory
             // (A memory dialog comes to the picture again)
-            IE::Blob::Ptr this_blob = extractBlob(ctx, i, views);
+            IE::Blob::Ptr this_blob = extractBlob(*ctx, i, views);
             iec.this_request.SetBlob(uu.params.input_names[i], this_blob);
         }
-        iec.this_request.Infer();
-        for (auto i : ade::util::iota(uu.params.num_out)) {
-            // TODO: Think on avoiding copying here.
-            // Either we should ask IE to use our memory (what is not always the
-            // best policy) or use IE-allocated buffer inside (and pass it to the graph).
-            // Not a <very> big deal for classifiers and detectors,
-            // but may be critical to segmentation.
 
-            cv::Mat& out_mat = ctx.outMatR(i);
-            IE::Blob::Ptr this_blob = iec.this_request.GetBlob(uu.params.output_names[i]);
-            copyFromIE(this_blob, out_mat);
-        }
+        iec.this_request.SetCompletionCallback(PostOutputs{iec, uu, out, sync, std::move(ctx)});
```

Comment:
so can it be lambda instead? You'd save a lot by implicitly capturing the scope.

Reply:
You can't move context to the lambda scope in c++11.

Reply:
PostOutputs/PostOutputList is reusable.

Reply:
Got it